cmd/query/app/querysvc/query_service.go

```diff
@@ -125,6 +131,32 @@ func (qs QueryService) GetDependencies(ctx context.Context, endTs time.Time, loo
 	return qs.dependencyReader.GetDependencies(ctx, endTs, lookback)
 }
 
+// MetricsQueryEnabled returns whether if metric query capabilities are enabled, opted-in by setting the
+// METRICS_STORAGE_TYPE environment variable.
+func (qs QueryService) MetricsQueryEnabled() bool {
```

Comment:
Why is this needed as public? For tests you can check qs.options directly

Reply:
To allow the HTTP and gRPC handlers to return early if the metrics query endpoints are called but not enabled like so:

HTTP Handler:
```
if !aH.queryService.MetricsQueryEnabled() {
	aH.handleError(w, errMetricsQueryDisabled, http.StatusNotImplemented)
	return
}
```

GRPC Handler:
```
if !g.queryService.MetricsQueryEnabled() {
	return nil, status.Error(codes.Unimplemented, "metrics querying is currently disabled")
}
```

An alternative could be to remove this endpoint and have the `QueryService.GetLatencies`, etc. functions check for `nil` `MetricsReader` then return a `MetricsReaderDisabled` error type and use [`errors.Is`](https://golang.org/pkg/errors/#Is) to handle the "not implemented" case. 

What do you think?

Reply:
@yurishkuro  I've moved metrics-related endpoints into a new `MetricsQueryService` and also added nil checks from each endpoint.